src/Microsoft.NetCore.Analyzers/UnitTests/ImmutableCollections/DoNotCallToImmutableCollectionOnAnImmutableCollectionValueTests.Fixer.cs

```diff
@@ -29,5 +32,199 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
             return new CSharpDoNotCallToImmutableCollectionOnAnImmutableCollectionValueFixer();
         }
+
+        public static readonly TheoryData<string> CollectionNames_Arity1 = new TheoryData<string>
+        {
+            nameof(ImmutableArray),
+            nameof(ImmutableHashSet),
+            nameof(ImmutableList),
+            nameof(ImmutableSortedSet)
+        };
+
+        public static readonly TheoryData<string> CollectionNames_Arity2 = new TheoryData<string>
+        {
+            nameof(ImmutableDictionary),
+            nameof(ImmutableSortedDictionary)
+        };
+
+        [Theory]
+        [MemberData(nameof(CollectionNames_Arity1))]
+        public void CA2009_Arity1_CSharp(string collectionName)
+        {
+            var initial = $@"
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+static class Extensions
```

Comment:
I believe it's not necessary to include source for extension methods, since they are already included in `ImmutableCollectionsSource.CSharp`

Reply:
@genlu it appears removing that makes the `ToImmutable` calls ambiguous. To avoid that, I'd probably have to pass `ReferenceFlags.RemoveImmutable` and update those `VerifyCSharpFix` and `VerifyBasicFix` methods to receive it.

Should I do that? Or should I just leave that class around?

Reply:
Let's use `ReferenceFlags.RemoveImmutable`, it seems the analyzer test uses this approach.